examples/interactive_local_hyperdrive_example.py

```diff
@@ -1,118 +1,228 @@
+# Alternatively, you can pass in one of the agents to use as the deployer. If you do this, ensure
+# there is enough base to cover the initial funding (defaults to 100_000 base.), and enough eth to cover gas.
 
 # Parameters for pool initialization. If empty, defaults to default values, allows for custom values if needed
-initial_pool_config = LocalHyperdrive.Config()
 # Launches 2 pools on the same local chain
-hyperdrive = LocalHyperdrive(chain, initial_pool_config)
-hyperdrive_2 = LocalHyperdrive(chain, initial_pool_config)
+
+# The config object is automatically build.
+# TODO add argument for deployer
+hyperdrive0 = LocalHyperdrive(chain)
+initial_pool_config = LocalHyperdrive.Config(initial_liquidity=FixedPoint(100_000))
+hyperdrive1 = LocalHyperdrive(chain, initial_pool_config)
 
 # %%
-# Generate funded trading agents from the interactive object
-# Names are reflected on output data frames and plots later
-agent0 = hyperdrive.init_agent(base=FixedPoint(100000), eth=FixedPoint(100), name="alice")
-hyperdrive_agent1 = hyperdrive_2.init_agent(base=FixedPoint(100000), eth=FixedPoint(100), name="bob")
-# Omission of name defaults to wallet address
-hyperdrive_agent2 = hyperdrive.init_agent(base=FixedPoint(100000), eth=FixedPoint(10))
 
-# Add funds to an agent
-agent0.add_funds(base=FixedPoint(100000), eth=FixedPoint(100))
+# We also add an active pool to this agent as well.
+# This allows us to mint base in initialization
+agent2 = chain.init_agent(base=FixedPoint(100_000), eth=FixedPoint(10), pool=hyperdrive0)
+
+# Add funds to an agent for various pools
+agent0.add_funds(base=FixedPoint(1_000_000), pool=hyperdrive0)
+agent1.add_funds(base=FixedPoint(100_000), pool=hyperdrive0)
+
+
+# %% [markdown]
+#####################
+# Executing Trades
+#####################
 
 # %%
-# Here, we execute a trade, where it's calling agent0 + gather data from data pipeline
-# under the hood to allow for error handling and data management
+# Here, we execute a trade on a pool.
 # Return values here mirror the various events emitted from these contract calls
-open_long_event_1 = agent0.open_long(base=FixedPoint(11111))
-open_long_event_1  # pyright: ignore
+# Here, base is unitless and is dependent on the underlying tokens the pool uses.
+open_long_event_1 = agent0.open_long(base=FixedPoint(11111), pool=hyperdrive0)
+
+# %%
+# We can also set an active pool an agent is using to avoid passing in the pool
+# for every trade.
+agent1.set_active(pool=hyperdrive0)
+open_long_event_2 = agent1.open_long(base=FixedPoint(22222))
+
+# Close previous longs from events
+close_long_event_1 = agent0.close_long(
+    maturity_time=open_long_event_1.maturity_time,
+    bonds=open_long_event_1.bond_amount,
+    pool=hyperdrive0,
+)
+
+# Close previous longs from the wallet
+# This wallet is a wallet object that contains positions relevant to a specific pool.
+# This takes a snapshot of the current wallet positions.
+agent0.get_wallet(pool=hyperdrive0)
```

Comment:
these `get_wallet` calls are going to create outputs that show the wallet info, right? If so, I think they should be their own cells so that the user can clearly see the different outputs for each agent. Then the closing events in its own cell.